Assets/HoloToolkit/GpuTiming/Scripts/GpuTiming.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+	/// <summary>
+	/// Encapsulates access to GPU timing methods.
+	/// </summary>
+	public static class GpuTiming
+	{
+		[DllImport("GpuTiming")]
+		private static extern IntPtr GetRenderEventFunc();
+
+		[DllImport("GpuTiming")]
+		private static extern double GetLastFrameGPUTime(int eventId);
+
+		private const int BaseBeginEventId = 1000;
+		private const int BaseEndEventId = 2000;
+
+		private static int nextAvailableEventId = 0;
+		private static Stack<int> currentEventId = new Stack<int>();
+		private static Dictionary<string, int> eventIds = new Dictionary<string, int>();
+
+		// Gets the latest available sample time for the given event.
+		public static double GetTime(string eventId)
```

Comment:
Was asked to preserve the double reported from the system at this level (arguably too much precision), since this can be used for high precision captures at runtime.